sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,247 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = "sale.order.import"
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    # Make doc_type field support for EDIFACT type
+    doc_type = fields.Selection(readonly=False)
+    parsed_json = fields.Text()
+    edifact_ok = fields.Boolean(compute="_compute_edifact_ok", store=True)
+    # Techincal fields used to get the extension part of files
+    file_ext = fields.Char(default="txt,d96a")
+    # EDIFACT format release
+    release = fields.Char(default="d96a")
+
+    @api.depends("order_file", "order_filename")
+    def _compute_edifact_ok(self):
+        for rec in self:
+            if not (rec.order_file and rec.order_filename):
+                rec.edifact_ok = False
+                continue
+            extensions = rec.file_ext.split(",")
+            path, ext = os.path.splitext(self.order_filename)
+            ok = ext and ext[1:] in extensions
+            if not ok:
+                ok = b64decode(self.order_file[:4]) == b"UNB"
+            rec.edifact_ok = ok
+
+    def button_parse_order_preview(self):
+        """
+        This method is called by the button to take a quick look at the pydifact
+        structure ('edifact' type) and the parsed object ('order') from
+        the original document before importing the document.
+        """
+        self.ensure_one()
+        edifact_model = self.env["base.edifact"]
+        order_file_decoded = b64decode(self.order_file)
+        parsed_obj = dict(edifact=None, order=None)
+        if self.edifact_ok:
+            parsed_obj["edifact"] = edifact_model.pydifact_obj(order_file_decoded)
+
+        parsed_order = self.parse_order(
+            order_file_decoded, self.order_filename, self.partner_id
+        )
+        del parsed_order["attachments"]
+        parsed_obj["order"] = parsed_order
+        self.write(
+            {
+                "parsed_json": json.dumps(parsed_obj, indent=4, default=str),
+            }
+        )
+        action = self.env["ir.actions.act_window"]._for_xml_id(
```

Comment:
Do you think returning the action is necessary for this feature? 
The purpose of the btn is to update the data for preview.... the wizard does not need any reload. 
Am I wrong?

Reply:
Yes, just update datda for preview.
What do you suggest, an @onchange instead?

Reply:
Yes, it's necessary. Because when we update the data for review, it will reload and make the wizard disappear. So we need the action to show the wizard again (after updating)

Reply:
You would change the 'Parse' button by 'Preview' and 'tech' status name to 'preview'.
In my opinion, i split 'parsed_json' to 'source_obj' and 'parsed_document', for example, in order to show next to each other in preview form. 
And a "Back" button in the preview step to go back to the 'import' state. 

And if @simahawk move this feature to the base module like he suggested before would be great.

Reply:
Thank @rmorant ,  @simahawk Do you think like Rmorant? And we should implement this feature here or `sale_order_import`?